NETDefault/WebApp/Controllers/CacheDependentController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Logging;
+
+namespace WebApp.Controllers
+{
+    /// <summary>
+    /// see Health check and application warmup
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CacheDependentController : ControllerBase
+    {
+        private readonly ILogger<CacheDependentController> _logger;
+        private readonly IMemoryCache _memoryCache;
+
+        public CacheDependentController(
+            ILogger<CacheDependentController> logger,
+            IMemoryCache memoryCache)
+        {
+            _logger = logger;
+            _memoryCache = memoryCache;
+        }
+
+        [HttpGet]
+        public ActionResult<string> GetString()
+        {
+            var value = _memoryCache.Get<string>("CacheItem");
+
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                _logger.LogInformation("Cache: ok");
+                return $"{value}";
+            }
+
+            _logger.LogInformation("Cache: not-ok");
+            return NotFound();
+        }
+
+        [HttpGet("check")]
+        public async Task<ActionResult> CheckCacheAsync()
+        {
+            do
+            {
+                _logger.LogInformation("Checking cache status..");
+
+                if (_memoryCache.TryGetValue<string>("CacheItem", out var _))
+                {
+                    _logger.LogInformation("Checking cache status: OK!");
+                    return Ok();
+                }
+
+                await Task.Delay(1000);
+            }
+            while (true);
+        }
```

Comment:
Maybe we can add an example route here in how to delete the cache (possibly by cahe key)?

Reply:
Same as the comment above. I think that if a project need a certain amount of cache management, that should be implemented for that project, and not be a one-size-fits-none solution in these example projects.

And busting cache in IMemoryCache when a site has multiple instances is quite hard. If that is a requirement, the cache should be moved to a Redis Cache, and that cache has quite an advanced CLI for those things.